operator/controllers/helmrelease_controller.go

```diff
@@ -294,6 +301,7 @@ func (r *HelmReleaseReconciler) rollback(ctx context.Context, rls *shipitv1beta1
 	}
 
 	r.Log.Info("rolling back HelmRelease", "release", releaseName)
+	r.notifier.Send(fmt.Sprintf("⚠️ `%s` is being rolled back.", releaseName))
```

Comment:
If `manager.Rollback` succeeds, but then the `Status().Update(...)` fails, we wouldn't notify slack. Same for all the other install/upgrade/delete etc. actions.

Not sure if it actually matters or if there's a proper solution that doesn't involve transactions 🙀, but we could avoid that scenario if slack notifications happened closer to the helm client calls: https://github.com/Wattpad/ship-it/blob/master/operator/controllers/release_manager.go#L16

Reply:
Yeah I thought about this but I think this is OK for now. If updating the status of a custom resource fails than that's likely to be a problem with kube itself and not something the developer is going to be able to action (other than to tell us). We'll probably want to spit out a log message at ERROR level and have an alert for it so velocity can investigate.